controllers/workspace/http.go

```diff
@@ -63,4 +72,41 @@ func setupHttpClients() {
 		Transport: healthCheckTransport,
 		Timeout:   500 * time.Millisecond,
 	}
+	InjectCertificates(k8s, logger)
+}
+
+func InjectCertificates(k8s client.Client, logger logr.Logger) {
+	if certs, ok := readCertificates(k8s, logger); ok {
+		for _, certsPem := range certs {
+			injectCertificates([]byte(certsPem), httpClient.Transport.(*http.Transport))
+		}
+	}
+}
+
+func readCertificates(k8s client.Client, logger logr.Logger) (map[string]string, bool) {
+	configmapRef := config.GetGlobalConfig().Routing.TLSCertificateConfigmapRef
+	if configmapRef == nil {
+		return nil, false
+	}
+	configMap := &corev1.ConfigMap{}
+	namespacedName := &types.NamespacedName{
+		Name:      configmapRef.Name,
+		Namespace: configmapRef.Namespace,
+	}
+	err := k8s.Get(context.Background(), *namespacedName, configMap)
+	if err != nil {
+		logger.Error(err, "Failed to read configmap with certificates")
+		return nil, false
+	}
+	return configMap.Data, true
+}
+
+func injectCertificates(certsPem []byte, transport *http.Transport) {
+	caCertPool := transport.TLSClientConfig.RootCAs
+	if caCertPool == nil {
+		caCertPool = x509.NewCertPool()
```

Comment:
According to [1]
```
	// RootCAs defines the set of root certificate authorities
	// that clients use when verifying server certificates.
	// If RootCAs is nil, TLS uses the host's root CA set.
```

@vinokurig  should we use SystemCertPool instead? 
```suggestion
		systemCertPool, err := x509.SystemCertPool()
		caCertPool = systemCertPool
```


[1] https://pkg.go.dev/crypto/tls

Reply:
Good catch @dkwon17.

I'm under the impression we should be using the SystemCertPool as well. If the DWOC is configured to use specific certs, a DevWorkspace that tries to resolve a URI  that uses another certificate will fail (when this should normally would work). For example, `https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-code/latest/devfile.yaml` can normally be resolved, but cannot when custom certs are configured with the functionality this PR introduces. 

To reproduce this, the following devworkspace can be used:

```YAML
apiVersion: workspace.devfile.io/v1alpha2
kind: DevWorkspace
metadata:
  name: code-latest
  namespace: openshift-operators
spec:
  contributions:
    - components:
        - container:
            env:
              - name: CODE_HOST
                value: 0.0.0.0
          name: che-code-runtime-description
      name: che-code
      uri: 'https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-code/latest/devfile.yaml'
  started: false
  template:
    commands:
      - exec:
          commandLine: echo "Hello from $(pwd)"
          component: dev
          workingDir: '${PROJECT_SOURCE}/app'
        id: say-hello
    components:
      - container:
          cpuRequest: 1000m
          image: 'quay.io/devfile/universal-developer-image:latest'
          memoryLimit: 512Mi
          memoryRequest: 256Mi
          sourceMapping: /projects
        name: dev
    projects:
      - git:
          remotes:
            origin: 'https://github.com/che-samples/web-nodejs-sample.git'
        name: web-nodejs-sample
```

The above devworkspace fails with:

```
status:
  conditions:
    - lastTransitionTime: '2024-05-22T19:45:49Z'
      message: 'Error processing devfile: failed to resolve component che-code by URI: failed to fetch file from https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-code/latest/devfile.yaml: Get "https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-code/latest/devfile.yaml": tls: failed to verify certificate: x509: certificate signed by unknown authority'
      reason: BadRequest
      status: 'True'
      type: FailedStart
```

If the call to `x509.SystemCertPool()` returns an error, we could either log the error and abandon trying to inject the certs, or we could create a new cert pool with  `x509.NewCertPool()` and log the error. The latter option is probably preferable.